R/call_upload_user_files.R

```diff
@@ -70,23 +131,31 @@ call_upload_user_files <- function(
   resource_filters = NA,
   resource_weight = NA,
   geo = "city",
-  acs_data_year = "2021"
-  ) {
-
-  #Define api URL
-  api_url <- stringr::str_glue("{base_url}/api/v1/upload-user-file/")
-
-  #Data Input Type Checks:
-  stopifnot(is.character(c(resource_file_path,
-                           resource_lat_column,
-                           resource_lon_column,
-                           geo,
-                           acs_data_year)
-                         )
+  acs_data_year = getOption("sedtR.year", 2021),
+  placement = "surface",
+  ...,
```

Comment:
I wasn't quite sure of the rationale for adding the `...` to this function - do you mind explaining how you'd envision it being used?

Reply:
Those dots get passed from `call_upload_user_files()` to `prep_sedt_resource()` and then to either `arcgislayers::arc_read()` *or* `sf::read_sf()`. Basically, it is a way of exposing parameters like `token` (for accessing private FeatureLayers) or `query` (for subsetting a file or database) that would only be used if a user is supplying an ESRI FeatureService URL or a file path. Most users probably wouldn't need this option but it could be really helpful if someone needs it.

The WIP book on Tidy design principles has more details on the uses of `...` arguments if it is a helpful reference: https://design.tidyverse.org/dots-after-required.html